src/PrimeTower.cpp

```diff
@@ -26,6 +26,13 @@ PrimeTower::PrimeTower()
 : wipe_from_middle(false)
 {
     const Scene& scene = Application::getInstance().current_slice->scene;
+
+    {
+        EPlatformAdhesion adhesion_type = scene.current_mesh_group->settings.get<EPlatformAdhesion>("adhesion_type");
+
+        multiple_extruders_on_first_layer = scene.current_mesh_group->settings.get<bool>("machine_extruders_share_nozzle") && ((adhesion_type == EPlatformAdhesion::RAFT) || (adhesion_type == EPlatformAdhesion::NONE));
```

Comment:
So basically if I understood correctly, this change is necessary because:

We normally only print the first layer prime tower with a single extruder, the adhesion extruder. This is done to improve stability of the prime tower. However that requires that all extruders are primed by the start g-code or the firmware at the start of the print. That assumption is dropped when extruders share a nozzle. Thus it is necessary for the prime tower to prime all extruders that are used on every layer, including the first one.

It would be good to document this, because it's pretty hard to figure this out reading the code later, especially for someone not involved with the [original issue](https://github.com/Ultimaker/Cura/issues/8148).

I think the fact that multiple extruders on the first layer are not allowed is a property of the skirt and brim, which both have extra skirt lines around the adhesion with the other extruders. So it'd be best to flip the check for the adhesion types around (with the same final result, until we add adhesion types).

Reply:
Yes, in fact I did not enable multiple extruders on the base layer of the prime tower in case the adhesion method is skirt or brim. I wanted to address only the problematic cases and not to change anything in the others.
I guess that your suggestion for 'documenting it' targets adding some comments in the code: I tried to use very informative variable names and kept the code readable, but I agree that adding some rationale could help in the future.
Unfortunately I'm still not used to this code review system, so ... I will do my best.